### ISSUE TITLE
AGray32, missing type parameter in superclass?

### ISSUE THREAD

**Issue by Evizero** [JuliaGraphics/ColorTypes.jl#44]
I am in the process of trying to increase test coverage and encountered a curious error when
trying to show a `AGray32` in the REPL

```julia
julia> AGray32(0x0duf8)
Error showing value of type ColorTypes.AGray32:
ERROR: MethodError: `show_ufixed` has no method matching show_ufixed(::Base.Terminals.TTYTerminal, ::ColorTypes.AGray32)
Closest candidates are:
  show_ufixed(::IO, ::ColorTypes.Colorant{FixedPointNumbers.UFixed{UInt8,8},1})
  show_ufixed{T,f}(::IO, ::ColorTypes.Colorant{FixedPointNumbers.UFixed{T,f},1})
  show_ufixed(::IO, ::ColorTypes.Colorant{FixedPointNumbers.UFixed{UInt8,8},2})
  ...
```

The closest candidates made me curious about `AGray32` base type so i looked and found

```julia
julia> super(super(super(AGray32)))
ColorTypes.Colorant{FixedPointNumbers.UFixed{UInt8,8},N}
```

which seems odd. shouldn't `N` be `2` or am I misunderstanding it? Looking at the [type definition](
https://github.com/JuliaGraphics/ColorTypes.jl/blob/master/src/types.jl#L329) is seems to me it was maybe forgotten to spell out.

Should I add it in the process?

**Comment by Evizero** [comment]
I think I am catching on. by storing both alpha and gray in a single variable, the generic way of redirecting component access tries to call `getfield(a,2)` if one were to set `N` to `2`. It seems to me the `AGray32` type is a weird outlier when it comes to component handling.

Maybe define custom methods where needed?

Or am I running in the wrong direction entirely?

**Comment by Evizero** [comment]
Setting `N` to 2 and defining `comp1(c::AGray32) = gray(c)` seems to fix apparent problems

**Comment by timholy** [comment]
You are indeed catching on! Your analysis seems exactly right. `AGray32` is definitely a special case.

**Comment by Evizero** [comment]
fixed in #47
